src/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,440 @@
+'use strict';
+
+const Base = require('./Base');
+const { Error } = require('../errors');
+const {
+  GuildScheduledEventEntityTypes,
+  GuildScheduledEventStatuses,
+  GuildScheduledEventPrivacyLevels,
+  Endpoints,
+} = require('../util/Constants');
+const SnowflakeUtil = require('../util/SnowflakeUtil');
+
+/**
+ * @type {WeakSet<GuildScheduledEvent>}
+ * @private
+ * @internal
+ */
+const deletedGuildScheduledEvents = new WeakSet();
```

Comment:
Hmm, since we're planning on removing this and `Structure#deleted`, should we just not implement it here? @discordjs/the-big-4

Reply:
I'm leaning towards not adding this at all, since it's going to be instantly deprecated. I'll be working on a PR in a bit to deprecate all `.deleted` later to prepare for the semver-major PR.

Reply:
aight, I'll remove these